mloop_multishot.py

```diff
@@ -13,6 +14,57 @@
 check_version('labscript_utils', '2.12.5', '3.0')
 
 
+def check_runmanager(config):
+    msgs = ['WARNING(s):']
+    rm_globals = rm.get_globals()
+    if not all([x in rm_globals for x in config['mloop_params']]):
+        msgs.append('Not all optimisation parameters present in runmanager.')
+    if not rm.get_run_shots():
+        msgs.append('Run shot(s) not selected in runmanager.')
+    if rm.error_in_globals():
+        msgs.append('Error in runmanager globals.')
+    n_shots = rm.n_shots()
+    if n_shots > 1 and not config['ignore_bad']:
+        msgs.append(
+            f'runmanager set to compile {n_shots:d} shots per request, but your '
+            + 'mloop_config has ignore_bad = False. You are advised to (i) remove '
+            + 'iterable globals so as to compile one shot per cost; or set ignore_bad '
+            + '= True in your mloop_config and only return one cost with bad = False '
+            + ' per sequence.'

```

Comment:
nitpick: Looks like there will be two spaces when these two lines are joined.

```suggestion
            + '= True in your mloop_config and only return one cost with bad = False '
            + 'per sequence.'
```